app/controllers/bot.js

```diff
@@ -1,140 +1,114 @@
 'use strict'
 require('dotenv').config()
 
-const { Client, RichEmbed } = require('discord.js')
-const sleep = require('sleep')
-const Sentry = require('@sentry/node')
-
-const discordService = require('../services/discord')
-
-const PermissionError = require('../errors/permission-error')
-const InputError = require('../errors/input-error')
-
-const commands = require('../commands')
+const path = require('path')
+const Guild = require('./guild')
+const Commando = require('discord.js-commando')
+const { RichEmbed } = require('discord.js')
+const SettingProvider = require('./settingProvider')
 
 const applicationConfig = require('../../config/application')
-const guildConfigs = require('../../config/guilds')
-
-const client = new Client()
 
-Sentry.init({ dsn: process.env.SENTRY_DSN })
-
-client.once('ready', async () => {
-    for (const [guildId, config] of Object.entries(guildConfigs)) {
-        const guild = client.guilds.find(guild => guild.id === guildId)
-        if (guild) {
-            const channel = guild.channels.find(channel => channel.id === config.rolesChannelId)
-            await channel.fetchMessages()
-        }
+module.exports = class Bot {
+    constructor() {
+        this.client = new Commando.Client({
+            commandPrefix: applicationConfig.defaultPrefix,
+            owner: applicationConfig.owner,
+            unknownCommandResponse: false,
+            disableEveryone: true
+        })
+        this.client.bot = this
+        this.client.setProvider(new SettingProvider())
+
+        this.client.registry
+            .registerGroup('admin', 'Admin')
+            .registerGroup('main', 'Main')
+            .registerDefaultGroups()
+            .registerDefaultTypes()
+            .registerDefaultCommands({
+                commandState: false,
+                unknownCommand: false,
+                ping: true,
+                help: true,
+                eval: true,
+                prefix: true
+            })
+            .registerCommandsIn(path.join(__dirname, '../commands'))
+
+        this.guilds = {}
+
+        this.client.once('ready', this.ready.bind(this))
+        this.client.on('guildMemberAdd', this.guildMemberAdd.bind(this))
+        this.client.on('messageReactionAdd', this.messageReactionAdd.bind(this))
+        this.client.on('messageReactionRemove', this.messageReactionRemove.bind(this))
```

Comment:
The messageReactionRemove event does not seem to fire. I will look into this in the following PR that updates the Discord.js dependency from v11 to v12.